pol-core/bscript/compiler/file/PrettifyFileProcessor.cpp

```diff
@@ -196,21 +197,163 @@ antlrcpp::Any PrettifyFileProcessor::visitVariableDeclarationList(
+    ++_currentgroup;
+    size_t curcount = linebuilder.currentTokens().size();
+    visitSequenceBindingList( ctx->sequenceBindingList() );
+    --_currentgroup;
+    addToken( "]", ctx->RBRACK(), linebuilder.closingBracketStyle( curcount ) );
+  }
+  else if ( auto lbrace = ctx->LBRACE() )
+  {
+    addToken(
+        "{", lbrace,
+        ( linebuilder.openingBracketStyle( false ) & ~FmtToken::ATTACHED ) | FmtToken::SPACE );
+    ++_currentgroup;
+    size_t curcount = linebuilder.currentTokens().size();
+    visitIndexBindingList( ctx->indexBindingList() );
+    --_currentgroup;
+    addToken( "}", ctx->RBRACE(), linebuilder.closingBracketStyle( curcount ) );
+  }
+
+  return {};
+}
+
+antlrcpp::Any PrettifyFileProcessor::visitBindingDeclarationInitializer(
+    EscriptGrammar::EscriptParser::BindingDeclarationInitializerContext* ctx )
+{
+  addToken( ":=", ctx->ASSIGN(), linebuilder.assignmentStyle() );
+  visitExpression( ctx->expression() );
+
+  return {};
+}
+antlrcpp::Any PrettifyFileProcessor::visitIndexBindingList(
+    EscriptGrammar::EscriptParser::IndexBindingListContext* ctx )
+{
+  auto args = ctx->indexBinding();
+  for ( size_t i = 0; i < args.size(); ++i )
+  {
+    visitIndexBinding( args[i] );
+
+    if ( i < args.size() - 1 )
+      addToken( ",", ctx->COMMA( i ),
+                linebuilder.delimiterStyle() | FmtToken::PREFERRED_BREAK | FmtToken::SPACE );
+  }
+
+  return {};
+}
+
+antlrcpp::Any PrettifyFileProcessor::visitSequenceBindingList(
+    EscriptGrammar::EscriptParser::SequenceBindingListContext* ctx )
+{
+  auto args = ctx->sequenceBinding();
+  for ( size_t i = 0; i < args.size(); ++i )
+  {
+    visitSequenceBinding( args[i] );
+    if ( i < args.size() - 1 )
+      addToken( ",", ctx->COMMA( i ),
+                linebuilder.delimiterStyle() | FmtToken::PREFERRED_BREAK | FmtToken::SPACE );
   }
+
+  return {};
+}
+
+antlrcpp::Any PrettifyFileProcessor::visitIndexBinding(
+    EscriptGrammar::EscriptParser::IndexBindingContext* ctx )
+{
+  if ( auto identifier = ctx->IDENTIFIER() )
+  {
+    make_identifier( identifier );
+  }
+
+  if ( auto ellipsis = ctx->ELLIPSIS() )
+  {
+    addToken( "...", ellipsis, FmtToken::SPACE | FmtToken::ATTACHED );
+  }
+  else if ( auto expression = ctx->expression() )
+  {
+    // Use `closingParenthesisStyle` to add FmtToken::SPACE to previous token
+    addToken( "[", ctx->LBRACK(),
+              ( linebuilder.closingParenthesisStyle( false ) & ~FmtToken::SPACE &
+                ~FmtToken::ATTACHED & ~FmtToken::BREAKPOINT ) );
+    visitExpression( expression );
+    addToken( "]", ctx->RBRACK(), FmtToken::ATTACHED );
+  }
+
+  if ( auto binding = ctx->binding() )
+  {
+    visitBinding( binding );
+  }
+
+  return {};
+}
+
+antlrcpp::Any PrettifyFileProcessor::visitBinding(
+    EscriptGrammar::EscriptParser::BindingContext* ctx )
+{
+  addToken( ":", ctx->COLON(), FmtToken::SPACE | FmtToken::ATTACHED );
```

Comment:
DelimiterStyle

Reply:
It's not a delimiter though, this is the `:` in `var { foo: bar }`.

Using delimterstyle makes it eg:
```
         lastcoord:
         { lastx:
           firstX,
           lasty:
           firstY,
           lastz:
           firstZ },
         firstRest... },
```
vs
```
         lastcoord: { lastx: firstX,
                      lasty: firstY,
                      lastz: firstZ },
```

Reply:
Yes due to breakpoint, for -> I used ` linebuilder.delimiterStyle() & ~FmtToken::BREAKPOINT & ~FmtToken::ATTACHED ` (basically just using the spacecfg entry.
Mmmh for structs := I used assignmentStyle.. not really consistent...